### PR TITLE
Add some convenience methods

### PR DESCRIPTION
**Pull request by jdfreder**
After this is merged we should publish

Merge #8 first

### CODE REVIEW

**Comment by jdfreder** [comment]
Now with unit tests

**Comment by rgbkrk** [comment]
Awesome. I've got one comment in here on the interface before we ship though.

File: index.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
+const uuid = require('node-uuid');
+
+/**
+ * Filter for response messages
+ * @param  {object}  msg - message to check
+ * @return {Boolean}
+ */
+function isChildMessage(msg) {
```
Comment (rgbkrk) [inline index.js:8]: I'm beginning to think this should be `isChildMessage(parentMessage, message)`, which we would use in an RxJS filter like so:

``` js
const isChildMessage = enchannel.isChildMessage.bind(null, msg);
const childMessages = stream.filter(isChildMessage)
```

File: test/index.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    map: () => shell,
+    subscribe: cb => cb.call(shell),
+  };
+  return channels = {
+    shell,
+    iopub: { complete: sinon.spy() },
+    stdin: { complete: sinon.spy() },
+    control: { complete: sinon.spy() },
+    heartbeat: { complete: sinon.spy() },
+  };
```
Comment (rgbkrk) [inline test/index.js:56]: Nice

**Comment by rgbkrk** [comment]
Complete with tests! You're on a roll!

File: index.js (hunk 1 above)
Comment (rgbkrk) [inline index.js:8]: The main reason is that we require a bind on `isChildMessage` though have no nice way of documenting that.

File: index.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+}
+
+/**
+ * Create a message
+ * @param  {string} username
+ * @param  {string} session  guid for the session
+ * @param  {string} msg_type
+ * @return {object}          msg
+ */
+function createMessage(username, session, msg_type) {
```
Comment (rgbkrk) [inline index.js:20]: Would we assume this always creates the latest version of the Jupyter messaging protocol?

**Pull request merged**